Assets/Scripts/FpsController.cs

```diff
@@ -0,0 +1,158 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FpsController : MonoBehaviour
+{
+    public Rigidbody body;
+
+    [Header("Camera Settings")]
+    public Camera cam;
+    public float lookHorizontalSensitivity = 1.0f;
+    public float lookVerticalSensitivity = 1.0f;
+    public float minimumPitch = -89.9f;
+    public float maximumPitch = 89.9f;
+
+    [Header("Jump Settings")]
+    public float jumpForce = 300.0f;
+    public float jumpInputBufferTime = 0.25f;
+
+    [Header("Movement Settings")]
+    [Range(0.1f, 15.0f)]
+    public float forwardSpeed = 5.0f;
+    [Range(0.1f, 15.0f)]
+    public float backSpeed = 3.0f;
+
+    private Vector2 _inputVector = Vector2.zero;
+    private bool _jumpInput = false;
+    private float _jumpTimeSinceInput = 0.0f;
+    private bool _grounded = false;
+
+    private float _lookYaw;
+    private float _lookPitch;
+
+    void Reset()
+    {
+        body = GetComponent<Rigidbody>();
+        cam = GetComponentInChildren<Camera>();
+    }
+
+    void Start()
+    {
+        Cursor.lockState = CursorLockMode.Locked;
+        _lookYaw = transform.eulerAngles.y;
+        _lookPitch = cam.transform.eulerAngles.x;
+    }
+
+    void Update()
+    {
+        Look();
+        MovementInput();
+    }
+
+    void FixedUpdate()
+    {
+        MoveLogic();
+        JumpLogic();
+
+        UpdateGroundedState();
+    }
+
+    void ApplyDirectionalSpeedMultipliers(ref Vector3 vector)
```

Comment:
You're using a mixure of using instance variables (_inputVector), using reference variables (like this method), and returning vectors from methods.

Suggest unifying them a little. Does `_inputVector` really need to be a field?

Reply:
I have taken the things you said into consideration. I went with removing those two methods entirely instead, as they are both very short and become understandable 1-2 lines of code when making use of the ternary operator.

I am using `_inputVector` to be able to follow the guideline of polling for input in Update, and then using it in FixedUpdate when it comes to physics-related things.